pkg/ctl/create/cluster.go

```diff
@@ -195,6 +197,17 @@ func doCreateCluster(cmd *cmdutils.Cmd, ngFilter *filter.NodeGroupFilter, params
 	if err := eks.ValidateFeatureCompatibility(cfg, kubeNodeGroups); err != nil {
 		return err
 	}
+
+	// Check if flux binary exists early in the process, so it doesn't fail at the end when the cluster
+	// has already been created with a missing flux binary error which should have been caught earlier.
+	// Note: we aren't running PreFlight here, we just check for the binary.
+	if cfg.HasGitOpsFluxConfigured() {
+		if _, err := exec.LookPath("flux"); err != nil {
+			logger.Warning(err.Error())
+			return errors.New("flux binary is required when gitops configuration is set")
```

Comment:
Based on the suggestion above:
```suggestion
			return fmt.Errorf("flux binary is required when gitops configuration is set: %w", err)
```

Reply:
Yeah sounds good